plugins/statistics.koplugin/main.lua

```diff
@@ -234,28 +248,112 @@ function getDatesForBookOldFormat(book)
     return generateReadBooksTable(book.title, dates)
 end
 
-function getDatesForBook(book)
+-- sdays -> number of days to show
+-- ptype -> daily - show daily without weekday name
+--          daily_weekday - show daily with weekday name
+--          weekly - show weekly
+--          monthly - show monthly
+function getDatesFromAll(sdays, ptype)
     local dates = {}
+    local sorted_tbl = {}
+    local diff
+    local book = {}
+    now_t = os.date("*t")
+    local from_begin_day = now_t.hour *3600 + now_t.min*60 + now_t.sec
+    local now_stamp = os.time()
+    local one_day = 24 * 3600 -- one day in seconds
+        for _, v in pairs(ReadHistory.hist) do
```

Comment:
wrong indentation for this for block? it should be on the same level as previous statement.